### PR TITLE
feat(devtools): create new storage preset by remote function

### PR DESCRIPTION
**Pull request by Asuka109**
## Summary

- Create new storage preset item and write it to config file.
- Watch on the changes of config files and syncing it to client.

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -83,5 +83,8 @@
     "**/.DS_Store": true,
     "**/Thumbs.db": true,
     "**/node_modules": false
+  },
+  "[javascript]": {
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
```
Comment (zllkjc) [inline .vscode/settings.json:88]: What's the purpose of this change. any docs ?

File: packages/devtools/plugin/src/cli.ts [hunk 2]
```diff
@@ -50,7 +52,27 @@ export const devtoolsPlugin = (
...
-        prepare: rpc.hooks.prepare,
+        async prepare() {
+          await rpc.hooks.prepare();
+
+          const appConfig = api.useConfigContext();
+          const appCtx = api.useAppContext();
+          const { devtools: options = {} } = appConfig;
+          updateContext(ctx, options);
+
+          const watcher = chokidar.watch(getConfigFilenames(), {
```
Comment (zllkjc) [inline packages/devtools/plugin/src/cli.ts:63]: do we need to close in restart hook

File: .vscode/settings.json (hunk 1 above)
Comment (Asuka109) [reply .vscode/settings.json:88]: Seems a unexpected change by vscode itself. I will revert it.

File: packages/devtools/plugin/src/cli.ts (hunk 2 above)
Comment (Asuka109) [reply packages/devtools/plugin/src/cli.ts:63]: Sure, I will do it.

**Comment by Asuka109** [comment]
@zllkjc Done there~

**Review by zllkjc — APPROVED** [review]
